lib/ansible/modules/system/systemd.py

```diff
@@ -457,6 +487,21 @@ def main():
                         (rc, out, err) = module.run_command("%s %s '%s'" % (systemctl, action, unit))
                         if rc != 0:
                             module.fail_json(msg="Unable to %s service %s: %s" % (action, unit, err))
+
+                        if module.params['detect_early_failure']:
```

Comment:
You can reproduce it for "stop" as well. If ExecStop is defined and fails, the service goes into the "failed" state (the daemon will still be terminated). Question is, do we want to report this as a failure or not.

I agree that it makes little sense for "restart", and neither for non-service units (e.g. mount units). so maybe restrict it to just starting and stopping of service units?